apps/src/lab2/views/components/AiTutor2Chat.tsx

```diff
@@ -100,6 +108,31 @@ const AiTutor2Chat: React.FunctionComponent<AiTutor2ChatProps> = ({
 }) => {
   const dispatch = useAppDispatch();
 
+  const [systemPrompt, setSystemPrompt] = useState<string>();
+
+  useEffect(() => {
+    if (customPromptName) {
+      fetchCustomPrompt(customPromptName).then(prompt => {
```

Comment:
What happens if the user starts using chat before the custom prompt has been set, say when there is a slow network?